examples/kafka-broker/docker-compose.yaml

```diff
@@ -0,0 +1,59 @@
+version: "3.8"
+services:
+
+  kafka-client:
+    build:
+      context: .
+      dockerfile: Dockerfile-kafka
+    restart: "no"
+    deploy:
+      replicas: 0
+
+  proxy:
+    build:
+      context: .
+      dockerfile: Dockerfile-proxy
+    depends_on:
+      kafka-server:
+        condition: service_healthy
+    ports:
+    - "${PORT_PROXY:-10000}:10000"
+    - "${PORT_ADMIN:-8001}:8001"
+
+  kafka-server:
+    build:
+      context: .
+      dockerfile: Dockerfile-kafka
+    depends_on:
+      zookeeper:
+        condition: service_healthy
+    healthcheck:
+      test: ["CMD", "kafka-topics", "--list", "--bootstrap-server", "localhost:10000"]
+      interval: 5s
+      timeout: 60s
+      retries: 120
+    environment:
+      KAFKA_BROKER_ID: 1
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+      # Listener value needs to be equal to cluster value in Envoy config
+      # (will receive payloads from Envoy).
+      KAFKA_LISTENER: PLAINTEXT://kafka-server:9092
+      # Advertised listener value needs to be equal to Envoy's listener
+      # (will make clients discovering this broker talk to it through Envoy).
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka-server:10000
```

Comment:
I'm afraid this might be "proxy:${port}", not "kafka-server", because we want clients to talk to proxy, the proxy is going to route client traffic to kafka-server

Reply:
yep, that is where we hit problems

if we set it to `proxy:1000` then it seems to go into a loop where kafka tries to check the proxy, but then the proxy fails because it calls kafka and finds it unhealthy

Reply:
even removing the healthcheck, and bringing it up manually hit the same problem

Reply:
seems like @nguyenptk figured out the problem (both the main kafka server and the broker need to be listed as listners) - really, really nice work @nguyenptk !!!

magically it also seems to have negated the need for the healthcheck - no warnings seem to be emitted (we might need to revisit this if we find it flakes in CI, but for now looking good)